ambry-server/src/main/java/com/github/ambry/server/AmbryRequests.java

```diff
@@ -508,6 +519,122 @@ public void handleDeleteRequest(NetworkRequest request) throws IOException, Inte
+        for (StoreKey storeKey : batchDeletePartitionRequestInfo.getBlobIds()){
+          blobDeleteStatuses.add(new BlobDeleteStatus((BlobId)storeKey, ServerErrorCode.Bad_Request));
+        }
+        // All blobs failed to delete
+        partitionResponseInfo = new BatchDeletePartitionResponseInfo(batchDeletePartitionRequestInfo.getPartition(), blobDeleteStatuses);
+        metrics.unExpectedStoreDeleteError.inc();
+      } finally {
+        long processingTime = SystemTime.getInstance().milliseconds() - startTime;
+        totalTimeSpent += processingTime;
+        publicAccessLogger.info("{} {} processingTime {}", batchDeleteRequest, response, processingTime);
+        // Update request metrics.
+        RequestMetricsUpdater metricsUpdater = new RequestMetricsUpdater(requestQueueTime, processingTime, 0, 0, false);
+        batchDeleteRequest.accept(metricsUpdater);
+        for (BlobDeleteStatus blobDeleteStatus: partitionResponseInfo.getBlobsDeleteStatus()){
+          // Set operation level error to Bad_Request if any of the blobs fails
+          ServerErrorCode error = blobDeleteStatus.getStatus();
+          if (error != ServerErrorCode.No_Error) {
+            // Handling for control test when request type is disabled + setting default to Bad_Request for any error.
+            operationServerErrorCode = error==ServerErrorCode.Temporarily_Disabled?ServerErrorCode.Temporarily_Disabled:ServerErrorCode.Bad_Request;
+            boolean logInErrorLevel = false;
+            if (error == ServerErrorCode.Blob_Not_Found) {
+              metrics.idNotFoundError.inc();
+            } else if (error == ServerErrorCode.Blob_Expired) {
+              metrics.ttlExpiredError.inc();
+            } else if (error == ServerErrorCode.Blob_Deleted) {
+              metrics.idDeletedError.inc();
+            } else if (error == ServerErrorCode.Blob_Authorization_Failure) {
+              metrics.deleteAuthorizationFailure.inc();
+            } else {
+              logInErrorLevel = true;
+              metrics.unExpectedStoreDeleteError.inc();
+            }
+            if (logInErrorLevel) {
+              logger.error("Store exception on a delete within a batch delete operation with error code {} for request {}", error, batchDeleteRequest);
+            } else {
+              logger.trace("Store exception on a delete within a batch delete operation with error code {} for request {}", error, batchDeleteRequest);
+            }
+          }
+        }
+        partitionResponseInfoList.add(partitionResponseInfo);
+
+      }
+    }
+    response = new BatchDeleteResponse(batchDeleteRequest.getCorrelationId(), batchDeleteRequest.getClientId(), partitionResponseInfoList, operationServerErrorCode);
+    logger.info(response.toString());
+    requestResponseChannel.sendResponse(response, request,
+    new ServerNetworkResponseMetrics(metrics.deleteBlobResponseQueueTimeInMs, metrics.deleteBlobSendTimeInMs,
```

Comment:
If you are planning to add metrics at batchdelete level, think its more appropriate to use them here?